xcp_d/utils/hcp2fmriprep.py

```diff
@@ -149,14 +210,34 @@ def hcpfmriprepx(hcp_dir, out_dir, subid):
         brainreg = pd.DataFrame(
             {"global_signal": gsreg, "white_matter": wmreg, "csf": csfreg, "rmsd": rsmd}
         )
+
+        # get derivatives and powers
         regressors = pd.concat([mvreg, brainreg], axis=1)
+        regressors["global_signal_derivative1"] = pd.DataFrame(
+            np.diff(regressors["global_signal"].to_numpy(), prepend=0)
+        )
+        regressors["global_signal_derivative1_power2"] = (
+            regressors["global_signal_derivative1"] ** 2
+        )
+        regressors["global_signal_power2"] = regressors["global_signal"] ** 2
+
+        regressors["white_matter_derivative1"] = pd.DataFrame(
+            np.diff(regressors["white_matter"].to_numpy(), prepend=0)
+        )
+        regressors["white_matter_derivative1_power2"] = regressors["white_matter_derivative1"] ** 2
+        regressors["white_matter_power2"] = regressors["white_matter"] ** 2
+        regressors["csf_power2"] = regressors["csf"] ** 2
+        regressors["csf_derivative1"] = pd.DataFrame(
+            np.diff(regressors["csf"].to_numpy(), prepend=0)
+        )
```

Comment:
I think `load_confounds` prepends the first value, rather than zero. Worth looking at later.

Reply:
I just copied the code from what XCP was using previously, but will make a note!